### ISSUE TITLE
[BUG] Hunter Pets dont have individual Cooldowns

### ISSUE THREAD

**Issue by Kenniflopper** [JamminL/cata-classic-bugs#1172]
#### 1.1. Description
Currently When you use cooldowns on Pet 1 and you dismiss the pet and summon Pet 2, their cooldowns are shared, the cooldowns should be individual.

 #### 1.2. How to Reproduce
        1. Tame 2 Monkey Pets
        2. Join an arena and use the special ability bad manner from Pet 1, then dismiss the pet and Summon Pet 2 and notice how the Bad manner cooldown carries over across the different pets.

Source Material:
https://www.youtube.com/watch?v=VrS1yty4ryY - Watch how i use Bad manner and Roar of sacrifice on Pet 1, then i dismiss pet and summon Pet 2 and that pets abilities are all on cooldown, cause its shared right now.

In original Cataclysm, Hunter pets would have their own individual cooldowns, so what hunters would do was rotate through different pets using the Dismiss Pet spell, effectively resetting their cooldowns. Hunters could do this until end of season 14 of Mop (Patch 5.4.2) a WHOLE expansion later, almost 2 expansions infact. Where they changed the way dismiss pet works, and removed it because it was no longer a mechanic that was needed inside arenas at that stage of the game for hunters. also note that in Wrath classic they removed pet cooldowns resetting upon dismissing and resummoning the SAME pet in pve / open world. But kept it for PvP play inside Arenas. This mechanic is not needed in PvE but affects PvP immensly for Hunters in Cataclysm.

Now the arguement will be, was this intended in Cataclysm or not? - My honest answer is yes, ofcourse it was as it was a widely known and used mechanic. There is a bluepost about it being removed in the end of MoP ( https://www.bluetracker.gg/wow/topic/us-en/10859655954-dismiss-pet-nerf/ ) But my arguement towards this is that it was only removed in Season 14 of Mop (Patch 5.4.2) near the end of a whole expansion later, almost 2 infact, and hunters worked completely different in MoP compared to Cataclysm, in Cataclysm Hunters were a CC / Support Bot that did no dmg outside of their burst openers with rapid fire and on use trinkets and lacked so many defensives, whereas in MoP hunters had everything. Self healing, Defensives, Damage, great crowd control without the use of pets. So i strongly believe this was the reason that they removed it so late cause when hunters had such a toolkit at their disposal, using the dismiss pet mechanic ontop of it made hunters completely broken in MoP.

Hunters in cataclysm rely heavily on this mechanic to compete at a somewhat decent level, we lack alot of defensives and we dont do damage outside our burst window, we have 1 form of hard cc which is freezing trap, which is easy to stop with any form of cc on the hunter.

Having this Mechanic is absolutely crucial for Hunters in Cataclysm and is a core part of our Kit and how they work in PvP, and sadly without it we do not function in a PvP environment.

If we start straying away from the original Cataclysm experience with such a massive change to the meta, where does it end? If this custom nerf to hunters will go through and completely remove them from the competitive scene, at what point do we stop nerfing other classes that suddently become super good and overpowered such as rogues and mages? Cataclysm PvP is balanced around the fact that hunters have the ability to do this in arena, and theres no denying that. As shown below it was widely used in a variety of tournaments and even from casual players.

  #### 2.2 Source Material (gonna get lengthy)
  
https://wowpedia.fandom.com/wiki/Patch_5.4.2
https://wowpedia.fandom.com/wiki/Dismiss_Pet

People also asking for Dismiss Pet changing macros during cata aswell, as it was widely used and known.
https://www.mmo-champion.com/threads/1165659-Hunter-macro-for-pet-changing?highlight=Monkey+pet+swap

TOURNAMENTS AND PVP VIDEOS:

https://www.youtube.com/watch?v=CaYHkdYeEbE - Focus on bottom right cds you will see a bunch of roar of sacrifice cds and Hunter's cast bar for dismiss pet cast:
GAME 1 = Sac used at 0:42 & Dismiss cast at 1:05 / New Sac used at 1:13 & Dismiss cast at 1:36 / New sac used at 1:39
GAME 2 = Sac used at 2:25 & Dismiss cast at 2:58 / Sac used 3:03 in middle of dismiss cast / Sac used at 3:15 & Dismiss cast at 3:33 / Sac used 4:04
GAME 3 = Sac used at 5:06 & Dismiss cast at 5:20 / Sac used at 5:29 & Dismiss cast at 5:31 / Sac used at 5:52 in middle of dismiss cast / Sac used at 6:11 & Dismiss cast at 6:27 / Sac used at 6:34 in middle of dismiss cast / Sac used at 6:43 in middle of dismiss cast / Sac used 6:51 in middle of dismiss cast / Sac used at 7:24 & Dismiss cast at 7:27
GAME 4 = Sac used at 8:45 & Dismiss cast at 9:15 / Sac used at 9:40 & Dismiss cast at 9:55 / Sac used at 10:05

https://www.youtube.com/watch?v=3jDr2QkDtfw - Focus on bottom left cds you will see a bunch of roar of sacrifice cds and Hunter's cast bar for dismiss pet cast:
GAME 1 = sac used at 2:45 & dismiss cast at 2:47 / sac used at 2:56 & pet abandon at 3:25 / sac used at 3:28 & dismiss cast at 3:49

Even famous players back then was doing it and uploading pvp montages to their youtube channels such as Braindeadly and Tosan.

https://www.youtube.com/watch?v=b1jk5nxa2nU - Braindeadly Hunter PoV: Notice when he cast dismiss pet and summons a different monkey the cooldowns are reset cause each monkey has their own cooldown timers
Timestamps = 3:20 & 3:46 & 6:21 & 8:01 & 11:32 & 12:24 & 15:37

Even at one point in tournaments Hunters would have 4 Roar of sacrifices on cooldown which means that they used the dismiss pet mechanic 4 times within the span of 1 minute to refresh their pet cooldowns.

![Tournament Pic 1](https://github.com/JamminL/cata-classic-bugs/assets/126105750/e648cc5b-4fe6-4c5e-b91b-cb0a94c688d0)
![Tournament Pic 2 Bottom right roar of sacrifice CD](https://github.com/JamminL/cata-classic-bugs/assets/126105750/46f3beb9-3b00-4161-9514-0480421fa75f)

**Comment by Pozzel** [comment]
This is a core mechanic of Hunter PvP, have also been on retail 12 years ago. This is, what made hunter as good as they are and awards a unique playstyle. Removing that would alter the game completely and in my opinion would ruin the feeling of playing classic. This is not some minor change like on wrath e.g. changing shadowmeld, no, this will alter the entire meta, giving a class, that RELIED on it to even be competitive in 3s, a huge disadvantage. People discussed this back and forth but I will stick to my original argument: YES it can be annoying, but there are ways to play around it and without it hunter will be in the gutter ones again, after being terrible all throughout wrath and tbc. I i think there are multiple githubs open regarding that topic and not much is there to be added, but removing that would basically kill the entire class in 3s.Or we all terrorize the ladder by playing 3 dps, fun fun fun.

**Comment by ManneN1** [comment]
This custom change, if it were to stay in, would in terms of how big of a nerf it is be equivalent to making Shadow Dance a 3min CD, or removing Mortal Strike.

I hope this is a bug that Blizzard is planning to fix asap, because this is on a completely different level which just screams: Hunter is not allowed to be S tier and the best class in PvP, classic is not classic - it's do whatever the majority wants (which after TBC and WotLK meta, are definitely not Hunter or Hunter comp players).

Priority on fixing this is needed, Aggrend and Blizzard.

**Comment by Kenniflopper** [comment]
I Assume something about that custom change in Wrath Classic where it would not reset in outdoor world / pve raids transferred over to Cataclysm and messed it up in PvP'd instances.

**Comment by skinnay-dev** [comment]
Need some communication from Blizzard on what their intention is. The assumption now is that the change is just naturally carrying over from WotLK without any consideration for how the pet system and the meta are different in Cataclysm.

**Comment by dedestus** [comment]
isnt this just round 2 of https://github.com/JamminL/cata-classic-bugs/issues/1076 ?

**Comment by Snipin** [comment]
> isnt this just round 2 of #1076 ?

Nope. That's a change request

This is a bug report for a mechanic that was in the game the entire expansion.

**Comment by Heroclassx** [comment]
Nothing I can add, what hasn't been said already. This mechanic has been in the original game back in original Cata 2011-2012.

I think noone is denying that it is a very strong mechanic and that is where the majority of complains are coming from, but this custom change would impact the pvp way more than any change which has been done on classic: wotlk and classic: tbc so far, it would turn the whole meta of the game, and in my opinion, not to the better but would make it drasticly more unfun for everyone involved.

**Comment by dedestus** [comment]
> > isnt this just round 2 of #1076 ?
> 
> Nope. That's a change request
> 
> This is a bug report for a mechanic that was in the game the entire expansion.

people here act like it is a discussion tho, so im going to repeat my statement as well. this has to stay out of the game for reasons you can read about in the original some changes suggestion

**Comment by Aggrend** [comment]
This is Not A Bug. We discussed this at length and spoke with people involved in hunter class design int he cata-mop timeframe and determined that this original functionality was always considered a bug, and was only not fixed during cata due to a technical limitation, which was resolved later on in wow's history. The WoW Classic Team generally has a general policy to try not to deliberately "rebreak" things that were clear bugs that would have been fixed when the content was current if time or technology had allowed. 

While we are sure some hunters may be disappointed, this is going to stay the way it is. As always, thank you for the detailed report and information on this issue.
